### PR TITLE
new(tests): add ASan to e2e tests

### PR DESCRIPTION
**Pull request by LucaGuerra**
**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

/kind feature

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

/area tests

**Does this PR require a change in the driver versions?**

No.

**What this PR does / why we need it**:

Add AddressSanitizer support to the e2e tests as well.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: .github/workflows/ci.yml
Comment (Molter73) [inline .github/workflows/ci.yml:283]: This is cool! I've been wanting to add it for some time now.

We might want to add `CMAKE_BUILD_TYPE=Debug` so we can get symbols for the ASAN reports. We could also add UBSAN, since it could catch some nasty undefined behaviour bugs and it works well with ASAN.
```suggestion
            cmake_opts: -DUSE_BUNDLED_DEPS=False -DCMAKE_C_FLAGS="-fsanitize=address -fsanitize=undefined" -DCMAKE_CXX_FLAGS="-fsanitize=address -fsanitize=undefined" -DCMAKE_BUILD_TYPE=Debug
```

We should also try to fix any memory leaks and/or undefined behavior that gets reported.

Comment (LucaGuerra) [reply .github/workflows/ci.yml]: Historically I've not had a lot of luck with ubsan, I set up the tests locally so I can try it out

Comment (LucaGuerra) [reply .github/workflows/ci.yml]: > We should also try to fix any memory leaks and/or undefined behavior that gets reported.

We can start from this one I think: https://github.com/falcosecurity/libs/pull/983 :)

Comment (LucaGuerra) [reply .github/workflows/ci.yml]: Yes, I can confirm, UBSAN segfaults for me even on regular unit tests 😭 needs further investigation

**Comment by LucaGuerra** [comment]
So @Molter73 to recap:
* After merging #983 ASan checks run fine :) Symbols are already resolved
* UBSAN does not work out of the box for me, and not necessarily due to UB faults but because of an unexpected segfault, meaning that it could either be an actual UB check that put the program in that state or issues with the instrumentation itself
* Performance looks good as the runtime difference with regular tests (amd64) is minimal
* I think we can merge this one and see if it is flaky or we identify other issues

**Review by Molter73 — APPROVED** [review]
/lgtm

Thanks for the hard work Luca! I'll try to look into UBSan on my end

**Review by jasondellaluce — APPROVED** [review]
/approve

**Pull request merged**
